### ISSUE TITLE
Getting blank white image when taking screenshot in Firefox.

### ISSUE THREAD

**Issue by mayurborkar27** [qq15725/modern-screenshot#27]
1. Install  modern-screenshot version - 4.4.25.
2. Using domToPng,  run project on Firefox.
3. After clicking on Screenshot getting white blank image.

After Clicking on take screenshot getting white blank image on Firefox, for Chrome it it working fine.
Image is getting  created properly.

- modern-screenshot: 4.4.25.
- Browser: Firefox : 113.0.2

**Comment by qq15725** [comment]
I need a minimal reproduction to debug your issue, thanks!

[Playground for https://codesandbox.io](https://codesandbox.io/s/modern-screenshot-lxhrev?file=/index.html)

[Playground for https://codepen.io](https://codepen.io/qq15725/pen/VwGeqwa)

[Playground for https://play.vuejs.org](https://play.vuejs.org/#eNplkk1rg0AQhv/KsFBqIFFpycWYQA899FAIpb3tZdGNmrof7I75IPjfO64mDeSguO++88y7417Ym7XxoZMsY7kvXGMRvMTObrhulDUO4QKlUd9mqyvoYeeMgmdlSun0gvxSal8bfL5zO7mbg9GfptMoy1sN9SAX14XRHsEZg+8trAd3NLuqpUDx8/XxL3N9A0XCn3UB0QzWG7hwDVd3fBBtJ6lGHEVDjClsNLYYd+eUyxeilRm8QE/g4cmT8cB0VFqgVLYVKGkFkJfNYciw5mzEcAZFK7yfBM6CjYzeCr3ZtuJcOcpZEnMQAiMhSGDTd6MqyLwrqH6KTUSP51aSgk5ovzNOZWPIKI2XMwjqkChaLNOnOQzv2YrKEoLmyV1eWgZUaBsP+cYBARybEusMXtPUnlajVMumqvFe68MoAoDN2fgfF0rYeO+NpnsRWHza8JxlVzpnDxdh2OWsRrQ+S5JO298qLoxKHow0HI9Jo0t5itWeqFOSnvV/3hTgRw==)

**Comment by mtxue97** [comment]
> @qq15725 we are having d3 chart as svg in our project and trying to take screenshot of that svg, in google chrome it is taking proper screenshot of that svg but in firefox it is getting as blank white image.

Is the image dataURL generated in Firefox very large and contains many redundant styles?

**Comment by henrikvilhelmberglund** [comment]
I saw this with some icons: [playground](https://play.vuejs.org/#eNrVV2tv2zYU/SuEgcAJFtIiKerhxgWWLl07IFv62D4ZKFRZtpXIkirKdrIi/32HpOQkTZqlaIFhHyzzPngvz73Uofh58HNds806G4wHRzpt8rolOmvX9fNpma/qqmnJZzKrVu+rs3JBrsm8qVZkuKpmWVNS+GdZqZdVO7zl3WTzQ1KVp9W6bLPZbg5ywGtaplWpW9JUVXtSkInx3j/otbOkTf58+/pGPS13gfYTfVWmZP+ATJ6Tz9OS9N5skxTrDHOSbZIjRrfYfZfCWQ+xLp0mRTYmglwjsPkdjRxgQIXQZqu6SNoMEiFHs3xj1jCZDlyY6YCkRaJ1p5gOrBscdZ2Uz8+K5GrRYJ0zxDSKzmii9NPmRXZ5M81Y2rwtMljeJ3mxzcvZrRwfclpUi0rTtrOlWtMcRfpQn1/KxRqBjkaI8VC4s6t2WZUPBaut5SshduN+ZMf5akHGukkRpis3Auv2ymZqm6TU86pZjV1x9z2mDojVmkruU+XtHRLzPHiGaSMEPRrdqjNEG8qmZaaurrGEbPNZuxwT6Xn15TOnWmb5Ytne1l27NTJT2X7iLNeIfjUmRnnH699K2kf4mKQXrpdjsm6KfQd8nK+SRTbSm8VPl6vi2bqdR4d78gVkssmz7XF1ORl6xCNCBYQrf9gvZDLMyyIvM/qxqNKLoQM2GfJsNewQdQKilnoyXLZtPR6Nttst20pWNYuR8DzP5B2SPXmClLNsrt3IxE2aX5tklmdlS/LZZGgR9vsJAF8Dn4fgfDKkgoVhtCcURIGknmfHVzBJ4YZQByFTKjCiy6HbqibVfA5OAEDrZlQ0rYqqmQz3hBQijo5/GY4e8O9zfDHDO/klPFb9jNFdGJ3yBmWdtEsyz4tiMjTtwPwHQR6g5JPhKRcR8d6EAcMfCXyiOBNvIskEOsPQGi+CRYo0ZqEMiWC+9AkPWBCHJGZ+HBPhM9/jhIdM8ugF9xULFSeBYJ4K4ekxESpi4/NY2MEnk8Nmo2b0ifK4S0d93/oxkVKbED1AQuoSQoWE1CWkfcLQZ8KPCfdZ6EtsJcwTCtEtsFMgsiltJuLRDuEnk4NahCalXcFTIUbcIuQSlQHEOLAIuZKmXEhqRz8WI0cVgTHmTAFjjBIDoq1ph2+3N7DvMZoODkhZ0Sars6Tt6ODmNaU6/xu8bjabfdx3sFtv7IipThrsss5nxxa3X9LO1lEQ97LVFwzUqSyx3NDKHXL9cWQiFF6g/5BM3GnScQgXLFbxjkPCmAXSiYZGYPV60VgjJjz+LVQio/DkeNf8p1CJDII49h+lkseB8Q5YzLDXd8BiD29fR4qwCpxr4Q2wKGK+sOJTgb18eeIp8S3AXr58ISX/To7serejxoDFeC29EK9qgLdWWgYB1MgnImKcq7tSYVyJsKVZBpxFQbSJUBj1yudoPH/HQI8EBqmUHZoMYZiCJhTzeUgkuATcKZkH7rktLAUC8A2VHuYGmnLGA0WdhwSpSOEEk1UGQkvwR6hwMiGMHVIpQSXxX/iLw3c89hkOLrPaWyBPY8EiDyrs0iDhyIGxfZp3C4yHMRjWPO9b6ROs7vm1yM7ad/DRDvGbwwtcDM5WPoIFad8k4tpCXVvuSgU1WKntUvgKPxZsqO3SMgqY73ONZvkRDgYWeoo4wfaZixTtN50KXLWpbc5dYUldq2ynlLaNcq2krlFOWNKuU7Trj+lVNyamV2Kj0BQuNVVMSKuLsClwriqOA9lKK/hzjqMjZip6oGNPrzr/nnb/Xw4fXDXsx/PgcODuXnSV1OxcVyXucvb8mXYGPR2M+xNpOrh3eTPW6cAcFhqnxbqsLxYsrVaje464GOh2hI+u7JKtzhG1+76+xhJaja+web74YgGIU+dF1vxRtzmueHcWkhRFtf3N6toGd7Reny6z9OIB/bnGBcos9azJdNZssulgZ2uTZpF1SE7e/Z5dYrwzAsi6gPcjxreZroq1WaNzO0brsOxbfna1r20983LxXp9ctqhKD8os1FbD+k8HuO2+eAT6zXIl83dVvP4HCzde4g==)

**Comment by PikPakPik** [comment]
> > @qq15725 we are having d3 chart as svg in our project and trying to take screenshot of that svg, in google chrome it is taking proper screenshot of that svg but in firefox it is getting as blank white image.
> 
> Is the image dataURL generated in Firefox very large and contains many redundant styles?

Yes same problem here with d3 chart

**Comment by PikPakPik** [comment]
> > @qq15725 we are having d3 chart as svg in our project and trying to take screenshot of that svg, in google chrome it is taking proper screenshot of that svg but in firefox it is getting as blank white image.
> 
> Is the image dataURL generated in Firefox very large and contains many redundant styles?

Yes very large and contains many redundant styles and same problem here with d3 chart, no solution since 1 year ?

**Comment by crazy4chrissi** [comment]
We had the same problem. The root cause were global css variables with embedded graphics that summed up to 8MB of variables at the global scope. It even caused modern-screenshot to crash (NS_ERROR_OUT_OF_MEMORY) in Firefox. Reducing the CSS size to ~ 50KB caused it to produce blank images in Firefox.

We solved this by moving the css variables to the scope of the element using it. After that, modern-screenshot did not have any issues producing the image anymore.

Before:
```
:root {
  --icon-foo: url(data:image/png;base64,..............................);
}
```
after:
```
my-icon {
  --icon-foo: url(data:image/png;base64,..............................);
}
```

In our case `my-icon` is a custom element with a shadow dom, thus the variable needs to be declared at the scope of the custom element (or a `::part`). If you are not using shadow dom, you can use css classes as selectors or whatever. Just make sure the dom elements in the tree that modern-screenshot should take images of do not have too many huge variables in their scope.
